keps/sig-cli/kustomize-subcommand-integration.md

```diff
@@ -0,0 +1,387 @@
+---
+title: Enable kustomize in kubectl
+authors:
+ - "@Liujingfang1"
+owning-sig: sig-cli
+participating-sigs:
+ - sig-cli
+reviewers:
+ - "@liggitt"
+ - "@seans3"
+ - "@soltysh"
+approvers:
+ - "@liggitt"
+ - "@seans3"
+ - "@soltysh"
+editors:
+ - "@pwittrock"
+creation-date: 2018-11-07
+last-updated: 2019-01-17
+status: implementable
+see-also:
+ - "[kustomize](https://github.com/kubernetes-sigs/kustomize/blob/master/docs/workflows.md)"
+ - "kustomize-file-processing-integration.md" 
+replaces:
+ - "0008-kustomize.md"
+superseded-by:
+ - n/a
+---
+
+# Enable kustomize in kubectl
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+  * [Goals](#goals)
+  * [Non-Goals](#non-goals)
+* [Kustomize Introduction](#kustomize-introduction)   
+* [Proposal](#proposal)
+  * [UX](#UX)
+     * [apply](#apply)
+     * [get](#get)
+     * [delete](#delete)
+  * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+  * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+Link to tracking issue: kubernetes/enhancements#633
+
+See [KEP FAQ](kep-faq.md) for "why not as as plugin".
+
+## Summary
+
+[Kustomize](https://github.com/kubernetes-sigs/kustomize)
+was developed as a subproject of sig-cli by kubectl maintainers with the goal of addressing
+a collection of issues (See [motivation](#motivation)) creating friction for declarative workflows in kubectl
+(e.g. `kubectl apply`).  The
+goal of the kustomize subproject was to bring this functionality back to kubectl to better complement
+`kubectl apply` and other declarative workflow commands.
+
+- declaratively generating Resource config
+- declaratively transforming Resource config
+- composing collections of Resource config across files and directories
+- layering the above on top of one another
+
+It is independent of, but complementary to, the [*server-side apply*](https://github.com/kubernetes/enhancements/issues/555)
+initiative that was started later and targeted at a separate collection of `kubectl apply` issues.
+
+**Note:**
```

Comment:
This paragraph (the "Note") and the next offer more details on the purpose / history of kustomize,

Trying to rephrase without the "Note":

> Kustomize offers generators and transformations in a _declarative_ form that improve
> on functionality provided by existing _imperative_ commands in kubectl.
>
> The declarative approach offers a clear path to accountability (all input can be kept
> in version control), can _safely_ exploit a holistic, unbounded view of disparate resources
> and their interdependence (it's a _plan_ about what to do, not a direct action), and
> can be easily constrained to verifiable rules across this view (all edits must be
> structured, no removal semantics, no environment side-effects, etc.).
>
> Imperative kubectl commands / flags available through kustomize:
>  ....

Reply:
Done